src/main/java/io/knowledgeassets/myskills/server/community/command/CommunityUserCommandController.java

```diff
@@ -71,11 +77,14 @@ public CommunityUserCommandController(CommunityUserCommandService communityUserC
 			@ApiResponse(code = 500, message = "Error during execution")
 	})
 	@DeleteMapping(path = "/communities/{communityId}/users/{userId}")
-	@PreAuthorize("isAuthenticated() and isPrincipalUserId(#userId)")
+	@PreAuthorize("(isAuthenticated() and isPrincipalUserId(#userId)) or hasCommunityManagerRole(#communityId)")
 	public ResponseEntity<CommunityResponse> leaveCommunity(@PathVariable("communityId") String communityId,
```

Comment:
It is somewhat unusual that a `DELETE` endpoint of a REST API returns a resource. In this case it's also strange that it returns a `CommunityResponse` although the path refers to a community user.
If it helps you optimize the interaction between the web app and the server then it's OK and you can leave it like this. Otherwise, I'd prefer to stick with REST conventions as far as possible.

Reply:
It does help optimize the interaction between the web app and the server as the web app can get a community with its members right after a member was removed. And thanks to that the web app can display a community state immediately after one has left a community / has been kicked out from a community.

I agree that having OK (200) status code with a community response for this DELETE endpoint related to community users looks quite strange but it is reasonable (from my point of view) in this particular case for the reason I have described above.

Reply:
OK, I understand. You can leave it as it is at the moment. Depending on the decision we make regarding the API design of the community features we can still refactor that later on. If we opt to have a separate API endpoint to retrieve community members the web app has to adapt the way of loading the members list component anyway... We'll do that separately at a later point of time.